### DISCUSSION TITLE
Busy-wait behave is different between pthread_mutex_lock and usleep

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#21662] [Q&A]
Thread is staring its work after almost one minute! despite of returning to the Browser Main Event Loop.
Not clear to me why, but the solution is calling to `sleep(1).`

emcc 3.1.55

```
std::mutex global_MutexLoadAsync3DMLMesh;

void CallingByThread()
{
std::lock_guard<std::mutex> oLock(global_MutexLoadAsync3DMLMesh);	
// do something without access to the main thread.
}


void Render()
{
std::lock_guard<std::mutex> oLock(global_MutexLoadAsync3DMLMesh);	
sleep(1);//  without this line the CallingByThread above is execute after more than one minute.
// Do something and return to the Browser Main Loop.
}

voidm main()
{
  emscripten_set_main_loop(Render, 0, 0);
}


```

**Comment by sbc100** [comment]
Do you think you could turn this into a full example that we can run?  

I don't see how adding a `sleep(1)` could do anything but delay the release of the lock).

**Comment by aharondavid** [comment]
I realize that this cause by a long `while`  (100ms) in the `Render()` function which doesn't let the Thread to start working.
Actually I think that I can reproduce it also in other OS like Windows. so it is not an issue.
